privacyidea/lib/resolvers/LDAPIdResolver.py

```diff
@@ -764,15 +767,25 @@ def loadConfig(self, config):
         self.resolverId = self.uri
         self.authtype = config.get("AUTHTYPE", AUTHTYPE.SIMPLE)
         self.tls_verify = is_true(config.get("TLS_VERIFY", False))
-        # Fallback to TLSv1. (int: 3, TLSv1.1: 4, v1.2: 5)
-        self.tls_version = int(config.get("TLS_VERSION") or ssl.PROTOCOL_TLSv1)
-
+        # Fallback to TLSv1. (int: 3, TLSv1.1: 4, v1.2: 5, TLS negotiation: 2)
+        self.tls_version = int(config.get("TLS_VERSION") or DEFAULT_TLS_PROTOCOL)
+        if self.tls_version:
+            tls_version = self.tls_version
+            tls_options = TLS_OPTIONS_1_3 if tls_version == int(ssl.PROTOCOL_TLS) else None
+        else:
```

Comment:
I am not clear about this if-else branch.
In case of LDAPS or STARTTLS the line

    self.tls_version = int(config.get("TLS_VERSION") or DEFAULT_TLS_PROTOCOL)

will always contain a value.

In case of plain LDAP the self.tls_version will not be relevant anyways.
So I am not sure, if we need the if-else or if we need it in a different way.